### ISSUE TITLE
when the hash like '/#/main/sub' style, ratchet.js can't work

### ISSUE THREAD

**Issue by CoraJoe** [twbs/ratchet#789]
when the hash like '/#/main/sub' style, ratchet.js can't work

**Comment by Johann-S** [comment]
Can you provide a example of your bug please ?

**Comment by CoraJoe** [comment]
```
  var getModal = function (event) {
    var modalToggle = findModals(event.target);
    if (modalToggle && modalToggle.hash) {
      return document.querySelector(modalToggle.hash);
      
      // try{
      //   var domModal = document.querySelector(modalToggle.hash);
      //   return domModal;
      // }catch(e){
      //   console.log(e);
      // }
      
    }
  };
```


 return document.querySelector(modalToggle.hash); 
 because document.querySelector('/main/sub')  will be  error;

**Comment by Johann-S** [comment]
Yes because your selector have to respect the standars, look at this page : 
http://www.w3schools.com/jsref/met_document_queryselector.asp

I'm not sure ``/`` are allowed into ID

**Comment by CoraJoe** [comment]
I knew selector is wrong, my point is hash like  '/#/main/sub' , eg: angular's router style,
ratchet.js can't work, I thought that the hash shouldn't relate to selector,  forgive my pool english,  thanks a lot

**Comment by CoraJoe** [comment]
or selector didn't query all hash, just query the hash that has modal/popover/.... etc
